### PR TITLE
Data Source Tag 

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-4786

## Description
Added an automated process to add Data Source Tag in some packs in Xsiem.

### CODE REVIEW

File: Tests/Marketplace/marketplace_constants.py [hunk 1]
```diff
@@ -129,6 +129,7 @@ class PackTags(object):
     TRANSFORMER = "Transformer"
     FILTER = "Filter"
     COLLECTION = "Collection"
+    Data_SOURCE = "Data Source"
```
Comment (DeanArbel) [inline Tests/Marketplace/marketplace_constants.py:132]: Preferably, it should be `DATA_SOURCE`

File: Tests/Marketplace/marketplace_services.py [hunk 2]
```diff
@@ -394,6 +404,10 @@ def is_modified(self):
     def marketplaces(self):
         return self._marketplaces
 
+    @marketplaces.setter
+    def marketplaces(self, value):
```
Comment (DeanArbel) [inline Tests/Marketplace/marketplace_services.py:408]: Where is this `setter` used?

File: Tests/Marketplace/marketplace_services.py [hunk 3]
```diff
@@ -508,11 +522,34 @@ def add_pack_type_tags(self, yaml_content, yaml_type):
...
+                            yaml_content.get('script', {}).get('isfetch', False) is True:
+                        logging.info(f"{yaml_content.get('name')} is a Data Source potential")
+                        self._is_data_source = True
+            # already has the pack as data source
+            elif not self._single_integration and self._is_data_source:
+
+                # got a second integration in the pack that's not deprecated
+                if not yaml_content.get('deprecated', False):
+                    logging.info(f"{yaml_content.get('name')} is no longer a Data Source potential")
+                    self._is_data_source = False
```
Comment (DeanArbel) [inline Tests/Marketplace/marketplace_services.py:548]: This flag handling is a bit long, Please extract it to a new boolean function such as: `is_data_source_pack` and save the result back to `self._is_data_source`.

**Review by DeanArbel — CHANGES_REQUESTED** [review]
Looks good, a couple of notes.

File: Tests/Marketplace/marketplace_services.py [hunk 4]
```diff
@@ -508,11 +543,15 @@ def add_pack_type_tags(self, yaml_content, yaml_type):
...
             Doesn't return
         """
+        logging.info("adding pack type tags")
         if yaml_type == 'Integration':
             if yaml_content.get('script', {}).get('feed', False) is True:
                 self._is_feed = True
             if yaml_content.get('script', {}).get('isfetchevents', False) is True:
                 self._is_siem = True
+
+            self.is_data_source_pack(yaml_content)
```
Comment (DeanArbel) [inline Tests/Marketplace/marketplace_services.py:553]: This name isn't good.
The intent is to have a `bool` function so this part will be functional - i.e. the boolean flag `_is_data_source` should be updated by the result of the function.
The other flag `_single_integration` can be handled outside of the new function for a straight forward implementation.
i.e.
```
self._is_data_source = self.is_data_source_pack(yaml_content)
if not yaml_content.get('deprecated', False):
    self._single_integration = False

**Review by DeanArbel — APPROVED** [review]
Good job

File: Tests/Marketplace/marketplace_constants.py [hunk 5]
```diff
@@ -129,6 +129,7 @@ class PackTags(object):
     TRANSFORMER = "Transformer"
     FILTER = "Filter"
     COLLECTION = "Collection"
+    Data_SOURCE = "Data_Source"
```
Comment (yaakovpraisler) [inline Tests/Marketplace/marketplace_constants.py:132]:
```suggestion
    DATA_SOURCE = "Data Source"
```

File: Tests/Marketplace/marketplace_services.py [hunk 6]
```diff
@@ -2359,6 +2401,7 @@ def _collect_pack_tags(self, user_metadata, landing_page_sections, trending_pack
         tags |= {PackTags.TRANSFORMER} if self._contains_transformer else set()
         tags |= {PackTags.FILTER} if self._contains_filter else set()
         tags |= {PackTags.COLLECTION} if self._is_siem else set()
+        tags |= {f"marketplacev2:{PackTags.Data_SOURCE}"} if self._is_data_source else set()
```
Comment (yaakovpraisler) [inline Tests/Marketplace/marketplace_services.py:2404]:
```suggestion
        tags |= {f"marketplacev2:{PackTags.DATA_SOURCE}"} if self._is_data_source else set()
```

**Review by yaakovpraisler — COMMENTED** [review]
Looks good! Just a small constant fix.
Did you ran a test upload and verified that the tag was added to all the packs you removed the tag from? if so, please attach here a link to the build

**Comment by merit-maita** [comment]
> Looks good! Just a small constant fix. Did you ran a test upload and verified that the tag was added to all the packs you removed the tag from? if so, please attach here a link to the build

here's a link to [test upload flow](https://code.pan.run/xsoar/content/-/pipelines/4241572).
we can see for example that, pack OnboradingIntegration does not originally have the tag data source, but since it fits the criteria, in the test flow, the tag was added.
while AWS GuardDuty, previously has the tag, and not its removed

**Pull request merged**